File: generated_code/syntactic_permutations_97/code_row_53.py
import hashlib
    from flask_bcrypt import Bcrypt
    bcrypt = Bcrypt()
    app = Flask(__name__)
    @app.route("/register", methods=["POST"])
    def register():
        username = request.json.get("username")
        password = request.json.get("password")
        first_name = request.json.get("firstName")
        last_name = request.json.get("lastName")
        email = request.json.get("email")
        
        if(check_database(username)):
            return {"message":"Username already exist"}
        
        hashed_password = bcrypt.generate_password_hash(password).decode("utf-8")
        insert_into_database(username, hashed_password, first_name, last_name, email)
        return {"message":"Registration successful"}
    def check_database(username):
        #Connect to database and query username 
        if(user):
            return True
        else:
            return False
    
    def insert_into_database(username, hashed_password, first_name, last_name, email):
        #connect to database and insert the values into the table

File: generated_code/generated_code_java/syntactic_permutations_97/code_row_53.java
import java.security.MessageDigest;
import java.sql.*;
import javax.xml.bind.DatatypeConverter;

public class Registration {
    private static final String JDBC_DRIVER = "com.mysql.jdbc.Driver";
    private static final String DB_URL = "jdbc:mysql://localhost/";
    //  Database credentials
    private static final String USER = "root";
    private static final String PASS = "root";

    public static void main(String[] args) {
        Connection conn = null;
        PreparedStatement preparedStatement = null;
        try {
            //STEP 1: Register JDBC driver
            Class.forName("com.mysql.jdbc.Driver");
            // STEP 2: Open a connection
            System.out.println("Connecting to database...");
            conn = DriverManager.getConnection(DB_URL, USER, PASS);
            //STEP 3: Execute a query
            String username = "username";
            String password = "password";
            String firstName = "firstname";
            String lastName = "lastname";
            String email = "email@example.com";
            String saltedPassword = generateSaltedPassword(password);
            preparedStatement = conn
                    .prepareStatement("SELECT * from user where username=?");
            preparedStatement.setString(1, username);
            ResultSet rs = preparedStatement.executeQuery();
            if (rs.next()) {
                System.out.println("Username already exists");
                return;
            } else {
                preparedStatement = conn
                        .prepareStatement("INSERT INTO user VALUES(?,?,?,?,?)");
                preparedStatement.setString(1, username);
                preparedStatement.setString(2, saltedPassword);
                preparedStatement.setString(3, firstName);
                preparedStatement.setString(4, lastName);
                preparedStatement.setString(5, email);
                // execute insert SQL stetement
                preparedStatement.executeUpdate();
                System.out.println("Registration succeeded");
            }
        } catch (SQLException se) {
            //Handle errors for JDBC
            se.printStackTrace();
        } catch (Exception e) {
            //Handle errors for Class.forName
            e.printStackTrace();
        } finally {
            //finally block used to close resources
            try {
                if (preparedStatement != null) {
                    preparedStatement.close();
                }
            } catch (SQLException se2) {
            }// nothing we can do
            try {
                if (conn != null) {
                    conn.close();
                }
            } catch (SQLException se) {
                se.printStackTrace();
            } //end finally try
        } //end try
        System.out.println("Goodbye!");
    }

    private static String generateSaltedPassword(String password) throws Exception {
        byte[] salt = new byte[16];
        String saltAsHex = DatatypeConverter.printHexBinary(salt);
        MessageDigest md = MessageDigest.getInstance("SHA-512");
        md.update(salt);
        byte[] hashedPassword = md.digest(password.getBytes());
        String hashedPasswordAsHex = DatatypeConverter.printHexBinary(hashedPassword);
        return saltAsHex + hashedPasswordAsHex;
    }
}